inc/Engine/Cache/PurgeExpired/Subscriber.php

```diff
@@ -54,11 +54,12 @@ public function __construct( Options_Data $options, PurgeExpiredCache $purge ) {
 	 */
 	public static function get_subscribed_events() {
 		return [
-			'init'                => 'schedule_event',
-			'rocket_deactivation' => 'unschedule_event',
-			static::EVENT_NAME    => 'purge_expired_files',
-			'cron_schedules'      => 'custom_cron_schedule',
-			'wp_rocket_upgrade'   => [ 'update_lifespan_option_on_update', 13, 2 ],
+			'init'                                   => 'schedule_event',
+			'rocket_deactivation'                    => 'unschedule_event',
+			static::EVENT_NAME                       => 'purge_expired_files',
+			'cron_schedules'                         => 'custom_cron_schedule',
+			'wp_rocket_upgrade'                      => [ 'update_lifespan_option_on_update', 13, 2 ],
+			'rocket_after_automatic_cache_purge_dir' => 'run_upgrade_routine',
```

Comment:
that won't work for everyone because you can disable automatic cache clearing when setting cache clean lifespan to zero

Reply:
so do u think that this may work if added to `after_rocket_clean_domain` for example or what do u suggest?

Reply:
Also what I'm trying to achieve is to fix this for most of customers, for sure we may miss some users with some configurations and for those, the main upgrade routine will happen when they open the admin page.